app/src/main/java/org/p2p/wallet/transaction/TransactionManager.kt

```diff
@@ -68,9 +68,12 @@ class TransactionManager private constructor(
         return executor.getStateFlow()
     }
 
-    fun getTransactionStateFlow(): Flow<TransactionState> = transactionStateFlow
-
-    fun getTransactionState(): TransactionState = transactionStateFlow.value
+    fun getTransactionStateFlow(refresh: Boolean = true): Flow<TransactionState> {
```

Comment:
refresh  -> forceIsProgress 
because you are force changing the value inside flow